### ISSUE TITLE
Variable expansion not working with mutations

### ISSUE THREAD

**Issue by dropje86** [hashicorp/nomad#1224]
Nomad v0.3.2
Consul v0.6.4
Description:	Ubuntu 14.04.4 LTS

I have a simple test job specification running a python web server. The service gets registered in Consul including a script health check. This works flawless except for variable expansion during "updates".
When I change the **group count** and perform `nomad run web.nomad` I get a service health check in Consul looking like this:

```
{
    "a4dfbee8062f8fc7be3fe2200c2c23810671b5e5": {
        "Node": "nomad",
        "CheckID": "a4dfbee8062f8fc7be3fe2200c2c23810671b5e5",
        "Name": "check_http_800${NOMAD_ALLOC_INDEX}",
        "ServiceName": "webservice",
        "Notes": "",
        "ModifyIndex": 0,
        "Status": "critical",
        "ServiceID": "nomad-registered-service-cd0c8730-75fa-3f3d-7aff-8f0a039cc3d1-web001-b5457a5a7acdb01c65b3afa23ffa66d0891b691b",
        "Output": "FAIL\n",
        "CreateIndex": 0
    }
}
```

The expansion of ${NOMAD_ALLOC_INDEX} (and probably other variables) doesn't happen for either the service name in consul and the actual command + args of that service (again, only during **updates**, the initial job run is doing it correctly).

Job file:

```
job "web" {
	region = "global"
	datacenters = ["dc1"]
	type = "service"
	priority = 50

	constraint {
		attribute = "${attr.kernel.name}"
		value = "linux"
	}

	update {
		# Stagger updates every 30 seconds
		stagger = "30s"

		# Update a single task at a time
		max_parallel = 1
	}

	group "webservers" {
		count = 1

		restart {
			attempts = 10
			interval = "1m"
			delay = "5s"
			mode = "fail"
		}

		task "web001" {
			driver = "exec"

			config {
				command = "/usr/bin/python"
				args = ["-m", "SimpleHTTPServer", "800${NOMAD_ALLOC_INDEX}"]
			}

			service {
				name = "webservice"
				tags = ["http", "python", "master"]
				check {
					type = "script"
					name = "check_http_800${NOMAD_ALLOC_INDEX}"
					command = "/usr/local/bin/health.sh"
					args = ["800${NOMAD_ALLOC_INDEX}"]
					interval = "5s"
					timeout = "1s"
				}
			}

			resources {
				cpu = 500
				memory = 64
				network {
					mbits = 10
				}
			}
		}
	}
}
```

**Comment by dadgar** [comment]
Nice find! Will get this fixed! Thanks

**Comment by diptanu** [comment]
@sean- Can you please verify whether your PR fixes this?

**Comment by diptanu** [comment]
I can see the codepath of updating the task is interpolating the service tags as well, so closing it. Fixed by #1201
